src/Workspaces/Core/Portable/Execution/ReferenceSerializationServiceFactory.cs

```diff
@@ -19,13 +19,13 @@ internal class ReferenceSerializationServiceFactory : IWorkspaceServiceFactory
         public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         {
             return new Service(
-                workspaceServices.GetService<ITemporaryStorageService>(),
+                workspaceServices.GetService<ITemporaryStorageService>() as ITemporaryStorageService2,
```

Comment:
I dont want it to required. 

whole purpose of this PR is
>this make serializer to work with less services.

not sure what you meant by making it required?

Reply:
>  it does not appear to be worth the additional complexity to allow down-stream clients to manually override the provided service for the sole purpose of removing it.

I in description explicitly said, i want to have temporary service removed. and this PR is so that it can be used without the service.

I opened an tracking issue saying this having trivia temproary service is just wrong.
https://github.com/dotnet/roslyn/issues/29111

Reply:
so this is why I didn't get TriviaTemporaryService. since that one doesn't implement ITemporaryStorageService2.